service-impl/data-service-impl/simple-data-service/src/main/java/uk/gov/gchq/palisade/data/service/impl/SimpleDataService.java

```diff
@@ -95,6 +96,8 @@ public SimpleDataService cacheService(final CacheService cacheService) {
     public CompletableFuture<ReadResponse> read(final ReadRequest request) {
         requireNonNull(request, "The request cannot be null.");
         //check that we have an active heartbeat before serving request
+        final String originalRequestId = UUID.randomUUID().toString();
```

Comment:
I would expect the read request to use an originalRequestId from the ReadRequest rather than starting a new chain